amads/core/vectors_sets.py

```diff
@@ -0,0 +1,294 @@
+# coding: utf-8
+"""
+Basic, shared functionality for sets and vectors.
+We aim for clear and consistent use of "set" and "vector" in  naming and documentation,
+though we tend to use basic classes such as lists in preference over the Python multiset class, for instance.
+
+Argument names are "set" or "vector" where that is the clear expectation and reason for the function.
+Where the function could be more widely applied, arguments are named `input_list`.
+"""
+
+from __future__ import annotations
+
+__author__ = "Mark Gotham"
+
+
+def set_to_vector(
+    input_set: list[int] | tuple[int],
+    min_index: int | None = 0,
+    max_index: int | None = 6,
+) -> tuple:
+    """
+    Converts any "set" (list of integers) into a "vector" (count of integers organised by index).
```

Comment:
It's a bit confusing that "set" doesn't actually accept a Python `set` here. I would suggested replacing the type hint for `input_set` with `Iterable[int]`.